ako-gateway-api/nodes/avi_model_l7_translator.go

```diff
@@ -79,12 +79,18 @@ func (o *AviObjectGraph) BuildChildVS(key string, routeModel RouteModel, parentN
 		}
 	}
 
+	// create vhmatch from the match
+	o.BuildVHMatch(key, childNode, rule, hosts)
+
+	if len(childNode.VHMatches) == 0 {
+		utils.AviLog.Warnf("key: %s, msg: No valid domain name added for child virtual service", key)
+		o.ProcessRouteDeletion(key, routeModel, fullsync)
```

Comment:
@pkoshtavmware yes this will also delete the objects because we initiating **ProcessRouteDeletion**. I think processing **o.BuildPGPool** etc should not matter in this case. If we do not initiate **ProcessRouteDeletion**, then the child vs is not deleted. Instead, ako-gateway tries to update the child vs with no domain name, there by resulting in error and ultimately, the child vs is not updated at all.

Reply:
Can this scenario be validated once manually since i do not see pool/poolgroup deletion within ProcessRouteDeletion ?
I.e. initially create a route with all valid domains then update it with all invalid domains and check pool/poolgroup deletion

Reply:
@pkoshtavmware I have already checked. Pool/PG deletion is processed after cache population once we do a PUT request for parent vs with updated checksum. It is similar to how child vs deletion is happening just that previously we were not checking the pool_group_ref while populating vs cache so pg and pools were not deleted along with the child vs.